custom_components/opensprinkler/__init__.py

```diff
@@ -150,7 +168,8 @@ async def async_added_to_hass(self):
 
     async def async_update(self):
         """Update latest state."""
-        await self._coordinator.async_request_refresh()
+        await self.hass.async_add_executor_job(self._controller.refresh)
+        self.async_write_ha_state()
```

Comment:
If we don’t use the coordinator I'm not entirely sure how the entities get notified that updated values are present. Does invoking this accomplish that?

Reply:
Yes